### DISCUSSION TITLE
Run scripts in parallel?

### DISCUSSION THREAD

**Discussion by ftzi** [oven-sh/bun#7452] [Q&A] [answered]
We have npm-run-all and concurrently. Does bun have a built-in feature that replaces these? That would be great.

**Comment by ramdanegie** [comment]
Bun doesn't have a built-in `--parallel` flag for `bun run`, but there are simple ways to run scripts in parallel:

**1. Shell background operator (simplest)**

```bash
bun run build:css & bun run build:js & wait
```

Works in bash/zsh, no dependencies needed.

**2. Use `Bun.$` in a script**

```ts
// scripts/parallel.ts
await Promise.all([
  Bun.$`bun run build:css`,
  Bun.$`bun run build:js`,
]);
```

Run with `bun run scripts/parallel.ts`.

**3. `concurrently` still works**

```bash
bun add -d concurrently
bunx concurrently "bun run build:css" "bun run build:js"
```

**Reply by ftzi** [reply]
Bro stop spamming AI answers in this repo

**Comment by RiskyMH** [comment, marked as answer]
@ramdanegie fyi that's not fully accurate as of bun v1.3.9 which has added support!:

https://bun.com/blog/bun-v1.3.9#bun-run-parallel-and-bun-run-sequential
